### ISSUE TITLE
Context shifting doesn't work in Mistral Large?

### ISSUE THREAD

**Issue by Vladonai** [LostRuins/koboldcpp#1095]
I am working with “magnum-v2-123b-Q4_K_L” model (I also tried “magnum-v2-123b-iQ4_K_M” - no difference). I've noticed that the context shift mechanism with this model works somehow wrong, if not to say it doesn't work at all. When I work with the “Llama3.1-70B_Q5_K_M” model with a context size of 16k and a response window of 1k tokens, I can delete even the last few replicas without having to recalculate the entire context. You could work for hours and not need to recalculate the context at all. But it's different with the Mistral Large. There, the context is often recalculated completely even when a new replica is added. Sometimes context shifting works - I haven't caught the pattern yet. This is very uncomfortable, especially when compared to the “Llama3.1-70B” model. Maybe there is a solution to the problem?

**Comment by LostRuins** [comment]
This is usually indicative of a tokenizer issue, i.e. the tokenizer is not tokenizing the tokens the same way depending on how they're arranged. For example, the gguf version of Llama3-8b used to have this issue: a newline combined with a different newline had different IDs depending on when they appeared.

**Comment by Vladonai** [comment]
> This is usually indicative of a tokenizer issue

Is this a problem with the model or the program? That is, is it possible to fix it programmatically or is the model hopeless?
[anthracite-org/magnum-v2-123b](https://huggingface.co/anthracite-org/magnum-v2-123b)

**Comment by markourm** [comment]
I am using koboldcpp backend and SillyTavern frontend and have not been successful in getting any of Mistral Large versions working, the original or finetunes. Have been using Q4_K_M gguf-s with partial offloading.

With short context it responds sensibly a first few messages, then degenerates quickly into repetitions and incoherence.
With starting context (long initial prompt) of 5000+ tokens, it outputs random characters. Have tried various combinations of settings and ST instruct formats, including those which have worked for other people.

Other models (mistral nemo, llama 3.1 70B, command-r-plus, etc) are all good with high context.

I'm wondering if it is a gguf issue affecting only Mistral Large.

**Comment by Vladonai** [comment]
> I am using koboldcpp backend and SillyTavern frontend and have not been successful in getting any of Mistral Large versions working, the original or finetunes.

I agree that the gguf version of this model has problems. I tried the exl2 version, for exllamav2, things are much better there (but there are other problems). Try in Silly Tavern under the AI Response configuration tab using the “MIrostat” preset. In Koboldcpp with this preset works and the model shows itself with the best results; interestingly, in llamacpp with the same preset after some time the model starts generating nonsense.

> I'm wondering if it is a gguf issue affecting only Mistral Large.

Problems with the context shift mechanism are also observed in the Qwen2-based Magnum model - there may be other problems with it as well.

**Comment by Vladonai** [comment]
Yes, those are definitely problems with this model (Magnum and any mix of it). I tried pure Mistral Large 123B - no problems, context shift works perfectly. I'll write to the author of the model about the problem, but I believe the algorithm for determining new parts of the prompt needs improvement...
